File: lending-backend/internal/model/item.go
package model

type Item struct {
	ObjectID      int    `json:"object_id"`
	ObjectName    string `json:"object_name"`
	Description   string `json:"description"`
	CurrentStatus string `json:"current_status"`
	ImageURL      string `json:"image_url,omitempty"`
}

type CreateItemRequest struct {
	ObjectName  string `json:"object_name"`
	Description string `json:"description"`
}

type UpdateItemRequest struct {
	ObjectName    string `json:"object_name"`
	Description   string `json:"description,omitempty"`
	CurrentStatus string `json:"current_status,omitempty"`
	ImageURL      string `json:"image_url,omitempty"`
}

type GetAllItemsResponse struct {
	ObjectID      int    `json:"object_id"`
	ObjectName    string `json:"object_name"`
	Description   string `json:"description"`
	CurrentStatus string `json:"current_status"`
	OwnerName     string `json:"owner_name"`
	OwnerEmail    string `json:"owner_email"`
	ImageURL      string `json:"image_url,omitempty"`
}

type GetItemMediaByItemID struct {
	Type         string `json:"type"`
	Creator      *string `json:"name"`
	Description  string `json:"description"`
	OriginalLink string `json:"link"`
	Media        string `json:"url"`
	CreatedAt    string `json:"created_at"`
}

File: lending-backend-dotnet/src/LendingSystem.Domain/Items/Item.cs
namespace LendingSystem.Domain.Items;

public static class ItemStatuses
{
    public const string Available = "Available";
    public const string OnLoan = "On Loan";
}

public sealed record Item(
    int ItemId,
    string ObjectName,
    string Maker,
    string Material,
    string Description,
    string CurrentStatus,
    string? ImageUrl);

public sealed record ItemSummary(
    int ItemId,
    string ObjectName,
    string Maker,
    string Material,
    string Description,
    string CurrentStatus,
    string? OwnerName,
    string? OwnerEmail,
    string? ImageUrl);

public sealed record ItemMediaSummary(
    string Type,
    string? Creator,
    string Description,
    string OriginalLink,
    string Media,
    DateTimeOffset CreatedAt);
